TIL.md

```diff
@@ -0,0 +1,263 @@
+# Thursday
+
+## Vim
+  - `$` - end of the line
+  - `0` - beginning of the line
+  - `c` - change
+  - `d` - delete 
+  - `DD` - remove whole line
+  - `P` - paste 
+  - `v` - visual mode
+  - `i` - insert mode
+  - `r` - replace
+  - `Esc` - returns to normal mode
+
+## Page object collection
+- collection option of the page objects allows to greatly simplify code
+
+```` javascript
+  products: collection({
+    itemScope: '[data-test-product-thumb]',
+    item: {
+      open: clickable('[data-test-product-thumb-link]'),
+      delete: clickable('[data-test-product-destroy-button]'),
+      cancelDelete: clickable('[data-test-cancel-button]'),
+      confirmDelete: clickable('[data-test-confirm-button]')
+    }
+  })
+````
+
+- functions on page objects can be chained
+
+```` javascript
+  await page.openProducts()
+    .openNewProduct()
+    .editProductName('FNX 45')
+    .saveProduct();
+````
+
+- `sinon.js` - used to create dummy functions (spies) and assert that they were called specific amount of times
+
+# Wednesday
+
+- TIL is important
+
+# Tuesday
+
+## ember-cli-sass
+
+## Useful testing packages
+https://github.com/cibernox/ember-native-dom-helpers
+https://github.com/simplabs/ember-test-selectors
+http://ember-cli-page-object.js.org/docs/v1.8.x/
+
+## Components
+- new components were added
```

Comment:
good to keep the list in order

1) router & route & template
2) component-name/{.hbs|.js}